bin/cassandra-ex

```diff
@@ -0,0 +1,228 @@
+# works before trying to use it (CASSANDRA-3245).
+NUMACTL_ARGS="--interleave=all"
+if which numactl >/dev/null 2>/dev/null && numactl $NUMACTL_ARGS ls / >/dev/null 2>/dev/null
+then
+    NUMACTL="numactl $NUMACTL_ARGS"
+else
+    NUMACTL=""
+fi
+
+if [ -z "$CASSANDRA_CONF" -o -z "$CLASSPATH" ]; then
+    echo "You must set the CASSANDRA_CONF and CLASSPATH vars" >&2
+    exit 1
+fi
+
+if [ -f "$CASSANDRA_CONF/cassandra-env.sh" ]; then
+    . "$CASSANDRA_CONF/cassandra-env.sh"
+fi
+
+# Special-case path variables.
+case "`uname`" in
+    CYGWIN*) 
+        CLASSPATH=`cygpath -p -w "$CLASSPATH"`
+        CASSANDRA_CONF=`cygpath -p -w "$CASSANDRA_CONF"`
+    ;;
+esac
+
+launch_service()
+{
+    pidpath="$1"
+    foreground="$2"
+    props="$3"
+    class="$4"
+    cassandra_parms="-Dlog4j.configuration=log4j-server.properties -Dlog4j.defaultInitOverride=true"
+
+    if [ "x$pidpath" != "x" ]; then
+        cassandra_parms="$cassandra_parms -Dcassandra-pidfile=$pidpath"
+    fi
+
+    # The cassandra-foreground option will tell CassandraDaemon not
+    # to close stdout/stderr, but it's up to us not to background.
+    if [ "x$foreground" != "x" ]; then
+        cassandra_parms="$cassandra_parms -Dcassandra-foreground=yes"
+        exec $NUMACTL "$JAVA" $JVM_OPTS $cassandra_parms -cp "$CLASSPATH" $props "$class"
+    # Startup CassandraDaemon, background it, and write the pid.
+    else
+        exec $NUMACTL "$JAVA" $JVM_OPTS $cassandra_parms -cp "$CLASSPATH" $props "$class" <&- &
+        [ ! -z "$pidpath" ] && printf "%d" $! > "$pidpath"
+        true
+    fi
+
+    return $?
+}
+
+# Parse any command line options.
+args=`getopt vfhp:bD:H:E: "$@"`
+eval set -- "$args"
+
+classname="org.apache.cassandra.service.CassandraDaemon"
+
+while true; do
+    case "$1" in
+        -p)
+            pidfile="$2"
+            shift 2
+        ;;
+        -f)
+            foreground="yes"
+            shift
+        ;;
+        -h)
+            echo "Usage: $0 [-f] [-h] [-p pidfile] [-H dumpfile] [-E errorfile]"
+            exit 0
+        ;;
+        -v)
+            "$JAVA" -cp "$CLASSPATH" org.apache.cassandra.tools.GetVersion
+            exit 0
+        ;;
+        -D)
+            properties="$properties -D$2"
+            shift 2
+        ;;
+        -H)
+            properties="$properties -XX:HeapDumpPath=$2"
+            shift 2
+        ;;
+        -E)
+            properties="$properties -XX:ErrorFile=$2"
+            shift 2
+        ;;
+        --)
+            shift
+            break
+        ;;
+        *)
+            echo "Error parsing arguments!" >&2
+            exit 1
+        ;;
+    esac
+done
+
+## A Little SED magic to doctor the classpath so that we use the
+## SNAPSHOT build in this directory rather than the "real" build found
+## by the included scripts above.
+echo "OLD_CLASSPATH:$CLASSPATH"
+CLASSPATH=$(echo $CLASSPATH | sed "s#:[^:]*apache-cassandra-\([^:]*\)2.0.17.jar:#:$(pwd)/apache-cassandra-\12.0.17-SNAPSHOT.jar:#g")
```

Comment:
Nope - sorry - I should've used git copy to start this script & the other, and *then* added these lines.